client/actions/userActions.js

```diff
@@ -0,0 +1,118 @@
+import axios from 'axios';
+import jwt from 'jsonwebtoken';
+import setAccessToken from '../utils/setAccessToken';
+import * as actionTypes from './actionTypes';
+import { beginAjaxCall } from './ajaxStatusActions';
+import handleError, { throwError } from '../utils/errorHandler';
+
+export function login(token, type) {
+  setAccessToken(token);
+  const decoded = jwt.decode(token);
+  const user = {
+    id: decoded.data.id,
+    roleId: decoded.data.roleId,
+    username: decoded.data.username
+  };
+  return {
+    type,
+    user
+  };
+}
+
+export function signup(signupDetails) {
+  return (dispatch) => {
+    dispatch(beginAjaxCall());
+    return axios.post('/users', signupDetails)
+      .then((res) => {
+        const token = res.data.token;
+        const tokenStorage = JSON.stringify({
+          jwt: token
+        });
+        localStorage.setItem('docman-pro', tokenStorage);
+        dispatch(login(token, actionTypes.LOGIN_SUCCESS));
+      })
+      .catch(error => handleError(error, dispatch));
+  };
+}
+
+export function signin(signinDetails) {
```

Comment:
Nitpick: signin should be signIn, signup === signUp.
Critical: signin and signup are doing exactly the same thing. DRY up the code please.